src/core/backbone_router/bbr_manager.cpp

```diff
@@ -78,10 +94,16 @@ void Manager::HandleNotifierEvents(Events aEvents)
     {
         if (Get<BackboneRouter::Local>().GetState() == OT_BACKBONE_ROUTER_STATE_DISABLED)
         {
+#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_MULTICAST_ROUTING_ENABLE
             Get<Tmf::TmfAgent>().RemoveResource(mMulticastListenerRegistration);
+#endif
+#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_DUA_NDPROXYING_ENABLE
             Get<Tmf::TmfAgent>().RemoveResource(mDuaRegistration);
+#endif
             mTimer.Stop();
+#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_MULTICAST_ROUTING_ENABLE
             mMulticastListenersTable.Clear();
```

Comment:
move this forward and bundle with removing mMulticastListenerRegistration resource?